### ISSUE TITLE
[Question] Why GatedSAETrainer do not implement resampling_neurons but GatedAnnealTrainer and StandardTrainer does?

### ISSUE THREAD

**Issue by jasonrichdarmawan** [saprmarks/dictionary_learning#53]
# Question

Is this to faithfully follow a paper?

For context: I am new to SAE, `sae_lens` library, and I have never tried `dictionary_learning` library

For context 2: I have a problem with `sparsity/dead_features` metric being too high after I change `sae_lens` library implementation to use `.detach()` in `via_gate_reconstruction = pi_gate_act @ self.W_dec + self.b_dec` [Code](https://github.com/jbloomAus/SAELens/blob/3432f0059bc1d90119fe48f88e17ac07aea3a620/sae_lens/saes/gated_sae.py#L196)

For context 3: I noticed that `sparsity/dead_feature` metric do not decrease over the course of the training and I can't find resampling neurons logic in `sae_lens` library. So, now I am looking at `dictionary_learning` implementation on how to implement resampling neurons

`GatedSAETrainer`

https://github.com/saprmarks/dictionary_learning/blob/60ec6bf5264944d64a4ca271f45a29ebfb9d4946/dictionary_learning/trainers/gdm.py#L13

`GatedAnnealTrainer`

https://github.com/saprmarks/dictionary_learning/blob/60ec6bf5264944d64a4ca271f45a29ebfb9d4946/dictionary_learning/trainers/gated_anneal.py#L105-L135

`StandardTrainer`

https://github.com/saprmarks/dictionary_learning/blob/60ec6bf5264944d64a4ca271f45a29ebfb9d4946/dictionary_learning/trainers/standard.py#L76-L109
